app/main/views/suppliers.py

```diff
@@ -107,16 +110,36 @@ def supplier_details():
     supplier['contact'] = supplier['contactInformation'][0]
     country_name = get_country_name_from_country_code(supplier.get('registrationCountry'))
 
-    framework = None
+    supplier_company_details_confirmed = supplier['companyDetailsConfirmed']
+    application_company_details_confirmed = None
+
+    frameworks = data_api_client.find_frameworks().get('frameworks')
+    framework_slug_map = {framework['slug']: framework for framework in frameworks}
+    open_framework_slugs = [framework['slug'] for framework in frameworks if framework['status'] == 'open']
+
+    unconfirmed_open_supplier_frameworks, unconfirmed_open_supplier_framework_names = [], []
+    if open_framework_slugs:
+        unconfirmed_open_supplier_frameworks = list(filter(
+            lambda sf: sf['frameworkSlug'] in open_framework_slugs and not sf['applicationCompanyDetailsConfirmed'],
+            data_api_client.get_supplier_frameworks(supplier_id=current_user.supplier_id)['frameworkInterest']
+        ))
+        unconfirmed_open_supplier_framework_names = [
+            framework_slug_map[sf['frameworkSlug']]['name'] for sf in unconfirmed_open_supplier_frameworks
```

Comment:
I may be being thick, but could you not just do `[sf['name'] for sf in unconfirmed_open_supplier_frameworks]` rather than going through the `framework_slug_map `?

Reply:
I really wanted to do that, but `unconfirmed_open_supplier_frameworks` is a list of SupplierFramework objects, which don't serialize the framework name back out - just the slug and family. So it's not available there sadly!

Reply:
Yep, that makes sense.